readthedocs-custom-steps/src/readthedocs_custom_steps.py

```diff
@@ -29,8 +29,25 @@ def main():
     print('readthedocs-custom-steps', __version__)
     return
 
-  filename = '.readthedocs-custom-steps.yml'
-  if os.path.isfile(filename):
+  def recursive_find_filename(fname, directory='.'):
+    for current_path, directories, files in os.walk(directory):
+      if os.path.split(current_path)[-1] == 'test':
+        continue
+      elif fname in files:
+        return os.path.join(directory, fname)
+      elif directories == '':
+        return None
+      else:
+        for next_directory in directories:
+          result = recursive_find_filename(directory=os.path.join(directory, next_directory), fname=fname)
+          if result:
+            return result
+        return None
+
+  filename = recursive_find_filename('.readthedocs-custom-steps.yml')
```

Comment:
I'm not a big fan of searching for the file anywhere in the repo. I'd much rather find an alternative solution to figuring out the configuration filename. Such as a comment in the readthedocs config? Or we expect it to live next to `requirements.txt`, so we can find the

```yaml
    requirements: docs/requirements.txt
```

key and construct the filename `docs/.readthedocs-custom-steps.yml`

Reply:
Three things made me feel uneasy as well -
- that a special case is required to prevent configurations files in a `test` directory
- that it might be possible to pick up a `.readthedocs-custom-steps.yml` within a venv in an unexpected way
- that a `print()` to stderr was needed to provide hints

So yes, there is too much potential for things to go wrong here, lets scrap it in favor of something that is directly user-defined.

I don't like the idea of using `.readthedocs.yml` with configuration in comments.  This means that commenting out this little configuration is probably not possible, you end up with an active configuration that you'd reasonably expect to be non-active.  Better to wait-it-out for readthedocs to not grok on `x-` prefixed configuration parameters.

My vote would be to add a check to find the `.readthedocs-custom-steps.yml` at the same path as the requirements as you've suggested - you'll need to keep checking for it at the top-level to support existing repos  - all round this feels like the best balance of options to me.